### PR TITLE
Checkout: Update credit card lists to use new CreditCard component

### PR DESCRIPTION
**Pull request by dechov**
Following up to https://github.com/Automattic/wp-calypso/pull/24485, introduces the new `<CreditCard>` component into the two contexts it was originally extracted from, and removes the original `<StoredCard>` from `my-sites/checkout`. No change in appearance or behavior from this refactor.

Checkout:

screen shot 2018-08-22 at 9 57 13 pm

Billing History:

screen shot 2018-08-22 at 10 42 34 pm

<s>To keep the appearance consistent with the Billing History view in `master`, a small change was also made to the component to remove 2px of left padding from the credit card container when not selectable (contrast the two screenshots). Note that this would result in misalignment when mixing in a single series `CreditCard` instances with and without `onSelect`.</s>

### CODE REVIEW

**Comment by dechov** [comment]
> Note that this would result in misalignment when mixing in a single series `CreditCard` instances with and without `onSelect`.

I've reverted this change, as I noticed the example in devdocs suffered from this very misalignment – as a result, the Billing History usage will have the same padding as the unselected Checkout options.

**Comment by dechov** [comment]
Now that https://github.com/Automattic/wp-calypso/pull/26898 is merged with slight modifications to the `CreditCard` component, this PR involves the visual changes from that one as well:

Before | After
-- | --
 |

**Comment by jsnajdr** [comment]
Hi @dechov :wave: This PR seems a bit forgotten, but it still seems very valuable and not obsolete at all, what do you think? I found it when working on CSS migration to webpack and when reviewing our usage of moment.js across Calypso.

I rebased it onto latest master, resolved the conflicts, tested it and applied a few more fixes. I think it's ready to get another round of review and testing and to be merged.

I migrated the CSS of the credit card related components to webpack. I also added the `withLocalizedMoment` HOC to a place where a moment.js instance was needed, as we plan to remove `moment` from the `i18n-calypso`'s `localize`.

There's also a little design change (related to accent colors) in the "new card" form when selecting the card to pay during checkout:

Before:
screenshot 2019-02-25 at 09 53 30

After:
screenshot 2019-02-25 at 09 55 18

I removed the accent color and instead made the header black and bold. Pinging @drw158 to reviews if my taste is any good :wink:

I also re-applied an AMEX fix from #28229 in 708d3e9. @ramonjd could you please have a quick look at that?

**Comment by dechov** [comment]
@jsnajdr I didn't forget but didn't see fit to prioritize – thanks for reviving and running with it! 👍

**Review by jsnajdr — APPROVED** [review]
I am approving (after thorough testing and review). But as I'm a co-author now, I'd like to get one additional :+1: before shipping.

**Comment by drw158** [comment]
Perfect thanks!

**Comment by ramonjd** [comment]
> I also re-applied an AMEX fix from #28229 in [708d3e9](https://github.com/Automattic/wp-calypso/commit/708d3e954d4ebbabfd213a3ae74683eb27d40ffa). @ramonjd could you please have a quick look at that?

LGTM 👍 

screen shot 2019-02-26 at 12 19 51 pm

**Review by ramonjd — APPROVED** [review]
I tested with various cards in checkout and the manage my cards section. LGTM

One area that I think should be improved in a future PR is allowing keyboard navigation for accessibility. 

As it is I can't select an existing CC via keyboard, or at least it's not obvious that I can do so.

I expected to be able to do this:

feb-26-2019 12-59-09

File: client/my-sites/checkout/checkout/credit-card-selector.jsx [hunk 1]
```diff
@@ -35,38 +36,44 @@ class CreditCardSelector extends React.Component {
 	}
 
 	storedCards = () => {
-		return this.props.cards.map( function( card ) {
-			const storedCard = <StoredCard card={ card } />;
-			return this.section( card.stored_details_id, storedCard );
-		}, this );
+		return this.props.cards.map( card => {
+			const onSelect = this.handleClickedSection.bind( this, card.stored_details_id );
```
[the remark this answers is not in the document]
Comment (jsnajdr) [reply client/my-sites/checkout/checkout/credit-card-selector.jsx]: Thanks for the review @ramonjd! Let's do the a11y and event handler improvements in a followup PR. This one is already quite big and the credit card code is revenue-sensitive, of course :)

**Pull request merged**